### PR TITLE
Endpoint to retrieve all future events for a user

### LINKED ISSUE #84: API: Endpoint to retrieve all the registered and future events for a specific user
Create an endpoint that retrieves all the events/study groups, that same user has registered for and are still active.

The endpoint should receive:

1. user id 
2. and the role has parameters 

and it should then filter

1.  the events that are active/ongoing/future, excluding canceled and past events/study groups, 

and return:

1. Event Id 
2. Organisation hosting the event/study group, 
3. Date of the event/study group
4. Time of the event/study group, 
5. Week number

and the list of : 

6.1. mentors attending if the requesting user is a student 
. **or** 
6.2 the list of students if the requesting user is a mentor.

The endpoint has to be documented on Swagger

### PR DESCRIPTION
**Pull request by AnilMula**
# Description
API: Endpoint to retrieve all the registered and future events for a specific user. At the moment it returns
1.Event Id
2.Organisation hosting the event/study group,
3.Date of the event/study group
4.Time of the event/study group,
5.Week number
and further has to work on 
6.1. mentors attending if the requesting user is a student
. or
6.2 the list of students if the requesting user is a mentor.

    

Fixes #84 

# How to test?

It can be  tested with swagger.

 note: in knex..select instead of today give any existing date in database
comment one of the statement to see the result
    if (role !== userRole) return ['incorrect role name in the url path'];
  
image
  

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/server/api/controllers/eventsUsers.controller.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      eventID: userEvent.eventID,
+      organization: userEvent.organization,
+      date: `${userEvents[0].dateTime.getDate()}-${
+        userEvents[0].dateTime.getMonth() + 1
+      }-${userEvents[0].dateTime.getFullYear()}`,
+      time: `${userEvents[0].dateTime.getHours()}:${userEvents[0].dateTime.getMinutes()}:${userEvents[0].dateTime.getSeconds()}`,
+      weekNumber: `${getNumberOfWeek(userEvents[0].dateTime)}`,
+    }));
+
+    if (userEvents.length === 0) return ['no matching records found'];
```
Comment (tpetrina) [inline src/server/api/controllers/eventsUsers.controller.js:72]:
```suggestion
    if (userEvents.length === 0) return [];
```
Never switch types at runtime! If there are no events found, the empty [] array means exactly that.

File: src/server/api/controllers/eventsUsers.controller.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    if (role !== userRole) return ['incorrect role name in the url path'];
+    if (role === 'student') {
+      return ['should return mentors attending'];
+    }
+    if (role === 'mentor') {
+      return ['should return students attending'];
+    }
```
Comment (tpetrina) [inline src/server/api/controllers/eventsUsers.controller.js:79]: Don't switch types! Do not return an array with a string inside if the default thing to return is an array of objects.
In the case of an error throw an `Error` or return `[]`!

**Review by tpetrina — CHANGES_REQUESTED** [review]
Good job, but small changes requested.

File: src/server/api/controllers/eventsUsers.controller.js (hunk 2 above)
Comment (AnilMula) [reply src/server/api/controllers/eventsUsers.controller.js]: removing lines from 74 to 79. as it is becoming complex to solve 6.1 and 6.2 from the issue.

**Review by MaherHussain — APPROVED** [review]

**Comment by MaherHussain** [comment]
nice

**Review by tpetrina — APPROVED** [review]
Looks good.

**Pull request merged**